vscode-runner: 1.6.1 -> 1.8.0

PhiliPdB opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
Updates vscode-runner to the latest upstream version ([v1.8.0](https://github.com/Merrit/vscode-runner/releases/tag/v1.8.0)).
Think this also closes: #378696 and #379081

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

PhiliPdB committed:
vscode-runner: 1.6.1 -> 1.8.0

Prince213 reviewed (approved):
Ran on x86_64-linux.

PhiliPdB commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 398130`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vscode-runner</li>
    <li>vscode-runner.pubcache</li>
  </ul>
</details>

pinage404 reviewed (approved):
Thanks !

It seems to works

---

When i run `nix-update vscode-runner --use-update-script`, i have only `package.nix` that is updated

I suppose that you did manual stuff to update `pubspec.lock.json`, right ?

Which command did you do ?

What do you think about adding this command to `passthru.updateScript` ?

PhiliPdB committed:
vscode-runner: Add update script

PhiliPdB commented:
Indeed the `pubspec.lock.json` needed updating. So I did that manually using the `yq` command mentioned in the docs. It seems wise to include this step in the `passtrhu.updateScript`.

I'm very new to Nix and NixOS, so I was not fully aware of all the 'automatic' update tools and possibilities. Looking around to see what the other Dart packages did, it seems we can use the same (or very similar) update script as the one used by [protoc-gen-dart](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/pr/protoc-gen-dart/update.sh). So that's what I added in the latest commit. I tested it on the current master branch, and it gives the exact same result compared to what I did manually.

pinage404 commented on pkgs/by-name/vs/vscode-runner/update.sh:
```diff
@@ -0,0 +1,36 @@
+
+# Get latest version number from GitHub
+version="$(
+  list-git-tags --url=https://github.com/Merrit/vscode-runner |
+    rg '^v(.*)' -r '$1' |
+    sort --version-sort |
+    tail -n1
```
In others packages, i've seen / done this : 

https://github.com/NixOS/nixpkgs/blob/70f9fa2d97fc93ce25782ce9ada6962a28d85cf8/pkgs/by-name/gk/gk-cli/package.nix#L106

https://github.com/NixOS/nixpkgs/blob/70f9fa2d97fc93ce25782ce9ada6962a28d85cf8/pkgs/by-name/ma/maestro/package.nix#L42

I don't know which methods is the best

PhiliPdB replied:
I like that this relies on Github to tell the latest version. Looking at the `protoc-gen-dart` package, the releases in their source repo seem more cluttered, which might explain this approach. But not sure of course.

Since I do like the suggested approach more, I'll rewrite it that way.

pinage404 commented on pkgs/by-name/vs/vscode-runner/update.sh:
```diff
@@ -0,0 +1,36 @@
+# Get latest version number from GitHub
+version="$(
+  list-git-tags --url=https://github.com/Merrit/vscode-runner |
+    rg '^v(.*)' -r '$1' |
```
for documentation and scripts, to improve readability and be more explicit, i recommend the long options when available

for example, in this case

```suggestion
    rg '^v(.*)' --replace '$1' |
```

pinage404 reviewed (commented):
Thanks to have adding this script !

---

I let minor suggestions

---

I tested the script on `master`

When i run `nix-update vscode-runner`, it only updates the `package.nix`

I removed the changes

When i run `nix-update vscode-runner --use-update-script`, it fails

```
$ nix eval --json --impure --expr 
let
  
  pkgs = import "/home/pinage404/Project/nixpkgs";
  args =  builtins.functionArgs pkgs;
  inputs = (if args ? system then { system = builtins.currentSystem; } else {}) //
           (if args ? overlays then { overlays = [ ]; } else {});
  pkg = (pkgs inputs)."vscode-runner";
  sanitizePosition = x: x;

  positionFromMeta = pkg: let
    parts = builtins.match "(.*):([0-9]+)" pkg.meta.position;
  in {
    file = builtins.elemAt parts 0;
    line = builtins.fromJSON (builtins.elemAt parts 1);
  };

  raw_version_position = sanitizePosition (builtins.unsafeGetAttrPos "version" pkg);

  position = if pkg ? isRubyGem then
    raw_version_position
  else if pkg ? isPhpExtension then
    raw_version_position
  else if (builtins.unsafeGetAttrPos "src" pkg) != null then
    sanitizePosition (builtins.unsafeGetAttrPos "src" pkg)
  else
    sanitizePosition (positionFromMeta pkg);
in {
  name = pkg.name;
  old_version = pkg.version or (builtins.parseDrvName pkg.name).version;
  inherit raw_version_position;
  filename = position.file;
  line = position.line;
  urls = pkg.src.urls or null;
  url = pkg.src.url or null;
  rev = pkg.src.rev or null;
  hash = pkg.src.outputHash or null;
  go_modules = pkg.goModules.outputHash or null;
  go_modules_old = pkg.go-modules.outputHash or null;
  cargo_deps = pkg.cargoDeps.outputHash or null;
  cargo_vendor_deps = pkg.cargoDeps.vendorStaging.outputHash or null;
  raw_cargo_lock =
    if pkg ? cargoDeps.lockFile then
      let
        inherit (pkg.cargoDeps) lockFile;
        res = builtins.tryEval (sanitizePosition {
          file = toString lockFile;
        });
      in
      if res.success then res.value.file else false
    else
      null;
  composer_deps = pkg.composerVendor.outputHash or null;
  composer_deps_old = pkg.composerRepository.outputHash or null;
  npm_deps = pkg.npmDeps.outputHash or null;
  pnpm_deps = pkg.pnpmDeps.outputHash or null;
  yarn_deps = pkg.offlineCache.outputHash or null;
  maven_deps = pkg.fetchedMavenDeps.outputHash or null;
  mix_deps = pkg.mixFodDeps.outputHash or null;
  tests = builtins.attrNames (pkg.passthru.tests or {});
  has_update_script = pkg.passthru.updateScript or null != null;
  src_homepage = pkg.src.meta.homepage or null;
  changelog = pkg.meta.changelog or null;
  maintainers = pkg.meta.maintainers or null;
} --extra-experimental-features flakes nix-command
$ nix-shell /home/pinage404/Project/nixpkgs/maintainers/scripts/update.nix --argstr package vscode-runner

Going to be running update for following packages:
 - vscode-runner-1.6.1

Press Enter key to continue...

Running update for:
Enqueuing group of 1 packages
 - vscode-runner-1.6.1: UPDATING ...
 - vscode-runner-1.6.1: ERROR

--- SHOWING ERROR LOG FOR vscode-runner-1.6.1 ----------------------

env: ‘/home/pinage404/Project/nixpkgs/pkgs/by-name/vs/vscode-runner/update.sh’: Permission denied


--- SHOWING ERROR LOG FOR vscode-runner-1.6.1 ----------------------
The update script for vscode-runner-1.6.1 failed with exit code 126
Traceback (most recent call last):
  File "/nix/store/3xakzxcxxlpvcycbad41r6n4xqcmixbl-nix-update-1.7.0/bin/.nix-update-wrapped", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "/nix/store/3xakzxcxxlpvcycbad41r6n4xqcmixbl-nix-update-1.7.0/lib/python3.12/site-packages/nix_update/__init__.py", line 311, in main
    package = update(options)
              ^^^^^^^^^^^^^^^
  File "/nix/store/3xakzxcxxlpvcycbad41r6n4xqcmixbl-nix-update-1.7.0/lib/python3.12/site-packages/nix_update/update.py", line 443, in update
    run(
  File "/nix/store/3xakzxcxxlpvcycbad41r6n4xqcmixbl-nix-update-1.7.0/lib/python3.12/site-packages/nix_update/utils.py", line 37, in run
    return subprocess.run(
           ^^^^^^^^^^^^^^^
  File "/nix/store/dzi9r3xn0ny4l4gwnyqpncljq40pdk6f-python3-3.12.8/lib/python3.12/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['nix-shell', '/home/pinage404/Project/nixpkgs/maintainers/scripts/update.nix', '--argstr', 'package', 'vscode-runner']' returned non-zero exit status 1.
```

How did you test? What did i do wrong?

PhiliPdB committed:
vscode-runner: improve update script

- get latest version from github releases instead of git tags
- use long option names in update script

PhiliPdB commented:
From the error log:
> env: ‘/home/pinage404/Project/nixpkgs/pkgs/by-name/vs/vscode-runner/update.sh’: Permission denied

How did you put the `update.sh` in master? To me this seems like it does not have execute permissions. As far as I can tell, git does have list the file with the correct permissions. I tested this by running `git ls-files -s pkgs/by-name/vs/vscode-runner/update.sh`.

I tested by manually setting the version, hash, and pubspec.lock.json to what is in master and let the update script run using `nix-update --build --use-update-script vscode-runner`. After running, the git diff didn't show any changes (except for the script addition).

pinage404 reviewed (approved):
I'm so sorry, i should have read the error log better, i focused on the python stack trace instead of reading just above it

khaneliman commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 398130`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vscode-runner</li>
    <li>vscode-runner.pubcache</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vscode-runner</li>
    <li>vscode-runner.pubcache</li>
  </ul>
</details>

khaneliman reviewed (approved):
Approved automatically following the successful run of `nixpkgs-review`.
